aws-android-sdk-appsync/src/main/java/com/amazonaws/mobileconnectors/appsync/AWSAppSyncClient.java

```diff
@@ -482,8 +568,20 @@ public AWSAppSyncClient build() {
                                 authMode.toString() + " but you selected " + selectedAuthMode.toString());
                     }
                 } catch (Exception exception) {
-                    throw new RuntimeException("Please check the AppSync configuration in awsconfiguration.json.", exception);
+                    throw new RuntimeException("Please check the AppSync configuration in " +
+                            "awsconfiguration.json.", exception);
+                }
+            }
+
+            // Handle case where the prefix is passed via code instead of awsconfiguration.json
+            // If mUseClientDatabasePrefix is true, a valid prefix is required
+            // Else do not use a prefix and use the default database names without a prefix
+            if (mUseClientDatabasePrefix) {
+                if (mClientDatabasePrefix == null) {
+                    throw new RuntimeException("Please pass in a valid ClientDatabasePrefix when useClientDatabasePrefix is true.");
                 }
+            } else if (mClientDatabasePrefix != null) {
```

Comment:
This logic is convoluted - and also repetitive as the same checks have been done earlier. Consider removing the idea of the "useClientDatabasePrefix" flag completely. That will simplify the logic here to "If there is a prefix, use it. Otherwise go with the defaults for the databases". Doing it this will will also eliminate any need to throw runtime exceptions

Reply:
Fixed the logic.